packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -182,11 +197,17 @@ def get_tx_in_db(session: Session, tx_sig: str) -> bool:
     return exists
 
 
-def get_purchase_metadata_from_memo(
+def parse_route_transaction_memo(
```

Comment:
Probably should have separate unit tests for this guy. non-blocking. I can add as a follow up test PR that i will work on this week :)

Reply:
Yeah not a bad idea. I think it's pretty well covered at the moment with the integration tests exercising the various cases:
* Transfer with no memo
* Transfer with purchase memo
* Transfer with recovery memo

I think the only one we don't have is a memo that's garbage/unrecognized and making sure that gets indexed as a standard transfer.